src/SFA.DAS.Commitments.Application/Commands/UpdateApprenticeship/UpdateApprenticeshipCommandHandler.cs

```diff
@@ -99,27 +124,39 @@ public sealed class UpdateApprenticeshipCommandHandler : AsyncRequestHandler<Upd
                           commitment.Id));
         }
 
-        private async Task CreateHistory()
+        private async Task UpdateStatusOfApprenticeships(List<Apprenticeship> apprenticeships, AgreementStatus agreementStatus)
         {
-            await _historyService.Save();
+            foreach (var apprenticeship in apprenticeships)
+            {
+                await _apprenticeshipRepository.UpdateApprenticeshipStatus(apprenticeship.CommitmentId, apprenticeship.Id, agreementStatus);
+            }
         }
 
-        private async Task UpdateStatusOfApprenticeship(Commitment commitment, Apprenticeship updatedApprenticeship)
+        private List<Apprenticeship> GetApprenticeshipsRequiringStatusUpdates(Commitment commitment, Apprenticeship updatedApprenticeship)
         {
+            var result = new List<Apprenticeship>();
+        
```

Comment:
I'm quite confused about this agreement status. And when it would be different from between apprentices from the same cohort. I would like to know why we do this (I know this is outside the scope of the current story).